### PR TITLE
Copy document type from canonical place when reindexing

### PR DESCRIPTION
**Pull request by suzannehamilton**
- When fetching all documents, look up the document `_type` from the top-level `_type` field, not the copy in `_source`. This is a step towards deleting `_source._type` entirely, which we need to do before upgrading to Elasticsearch 2.
- Move default `_type` assignment to POST request. Add more validation of search content `_type`, and move the fallback to "edition" to the /documents POST endpoint, which is the only place which is permitted to create documents without supplying a specific type. This should help us avoid situations like the recent corrupted search data, in which the `_type` was not being specified properly and was falling back to "edition" even for specialist document types.

https://trello.com/c/npsstuyq/103-elasticsearch-document-type-is-wrong-for-specialist-documents

Paired with @MatMoore.

Marked as DO NOT MERGE while we add more integration tests.

### CODE REVIEW

**Comment by suzannehamilton** [comment]
@tijmenb or @carvil - would one of you be able to review this? It should let us reenable the overnight analytics job (once I've tested it on staging).

**Review by carvil — APPROVED** [review]
This looks good to me

**Comment by suzannehamilton** [comment]
Cheers. I'm going to merge and deploy to staging for some more testing.

**Comment by carvil** [comment]
Ok, sounds good to me. You can also deploy the branch to staging if you prefer to do that before merging, I'm happy either way.

**Comment by suzannehamilton** [comment]
Good idea. I'll mark this as DO NOT MERGE while we're testing. 👍

**Pull request merged**
